hooks/useDragSelect.ts

```diff
@@ -0,0 +1,76 @@
+import { useState, useRef, useEffect, RefObject } from "react";
+
+interface UseDragSelectReturn {
+  draggedIndex: number[];
+}
+
+function useDragSelect<T extends HTMLElement>(
+  containerRef: RefObject<T>,
+  handleDragEnd: (draggedIndex: number[]) => void,
+): UseDragSelectReturn {
+  const [isDragging, setIsDragging] = useState(false);
+  const [startIndex, setStartIndex] = useState<number | null>(null);
+  const [endIndex, setEndIndex] = useState<number | null>(null);
+  const [draggedIndex, setDraggedIndex] = useState<number[]>([]);
+
+  useEffect(() => {
+    const handlePointerDown = (e: PointerEvent) => {
+      if (containerRef.current && containerRef.current.contains(e.target as Node)) {
+        const targetIndex = parseInt((e.target as HTMLElement).dataset.index || "-1", 10);
```

Comment:
```suggestion
        const targetIndex = parseInt((e.target as HTMLElement).dataset.index || "-1", 10);
```
p4: `dataset.index`에는 0도 있긴 한 것 같은데 문자열 `"0"`이라서 `falsy`한 값이 되지 않을 것 같네요 그래도 조금 더 명확하게 `??`연산자를 사용하는 것도 좋을 것 같아요!